### ISSUE TITLE
Composition data of draft enabled entity gets lost after draft activate

### ISSUE THREAD

**Issue by mengmengh2022** [SAP/open-ux-odata#456]
### Description
In our FE app, we are using draft enabled Odata v4 parent entity with several composition child entities. 
In these children entities, one entity stores the M:N relation of two other children entities. (see definition of **ChildEntity3** below)

In the display mode and after entering the draft mode (after clicking the Edit button), all data are displayed as expected.

**Error: without doing any change in the draft and when click on the Save ("draftActivate"), some of the children entities data gets lost.**

### Steps to Reproduce

1. Create a draft version of the FE draft enabled parent object (aka click Edit button)
2. See the data of childEntity1, childEntity2 and childEntity3 on UI as defined in local data files. (One row on each child-entity table, see local data definition below)
3. Do no change in the draft
4. Activate the draft (aka click Save button)
5. See that data on these child tables is missing. 
(On the table childEntity1 and childEntity2 no row is displayed.  On the table childEntity3 the table count shows "1" but the column values are displayed as empty string. )

### Expected results
The data should not get lost when activating draft.

### Actual results
The data gets lost when activating draft.

#### Version/Components/Environment	
@sap-ux/ui5-middleware-fe-mockserver version 2.1.62

#### Some other findings from the test:

- When saving draft: the GET request for childEntity3 does not select the field "HasActiveEntity". 
- When saving draft: in the response of GET request for childEntity3, the expand to relationToEntity1 and relationToEntity2 contain the one row but with empty value in fields "code" and "name". 
- When I remove the local data of childEntity3, restart the mockserver, enter draft and save the draft, the data does not get lost. The issue seems to be related to this childEntity3.

##### CDS Data model:

```
entity Application: cuid, managed {
  name     : localized String(100);
  childEntity1: Composition of many ChildEntity1 on childEntity1.application = $self;
  childEntity2: Composition of many ChildEntity2 on childEntity2.application = $self;
  childEntity3: Composition of many ChildEntity3 on childEntity3.application = $self;
}

entity ChildEntity1 : CodeList {
    key code          : types.Code;
        ...
        application   : Application;
}

entity ChildEntity2 : CodeList {
    key code          : types.Code;
        ...
        application   : Application;
}

entity ChildEntity3 : cuid {
    relationToEntity1  : Association to ChildEntity1;
    relationToEntity2  : Association to ChildEntity2;
    application        : Application;
}

```

##### Service CDS:

  ```
@odata.draft.enabled
  @Capabilities : {
    Insertable : false,
    Updatable  : true,
    Deletable  : false
  }
  entity Application                         as projection on namespace.Application;

  entity ChildEntity1 as projection on namespace.ChildEntity1;
  annotate namespace.ChildEntity1 with @fiori.draft.enabled;

  entity ChildEntity2 as projection on namespace.ChildEntity2;
  annotate namespace.ChildEntity2 with @fiori.draft.enabled;

  entity ChildEntity3  as projection on namespace.ChildEntity2;
```

##### Local service json data:

```
// Application
[
  {
    "ID": "6c1133ef-27c7-40d4-8c81-a9f9620bcaab",
    "name": "app"
  }
]

//ChildEntity1
[
  {
    "code": "C1",
    "name": "C1",
    "application_ID": "6c1133ef-27c7-40d4-8c81-a9f9620bcaab",
    ...
  }
]
//ChildEntity2
[
  {
    "code": "A1",
    "name": "A1",
    "application_ID": "6c1133ef-27c7-40d4-8c81-a9f9620bcaab",
    ...
  }
]
//ChildEntity3
[
  {
    "ID": "7c09a1dc-3f69-4ab5-aeef-02449ea3322e",
    "relationToEntity1_code": "C1",
    "relationToEntity2_code": "A1",
    "application_ID": "6c1133ef-27c7-40d4-8c81-a9f9620bcaab"
  }
]


```
##### Batch requets sent when opening the app in Display mode:

```
GET Applications(ID=6c1133ef-27c7-40d4-8c81-a9f9620bcaab,IsActiveEntity=true)/childEntity3?$count=true&$select=ID,IsActiveEntity,relationToEntity2_code,relationToEntity1_code&$expand=relationToEntity2($select=IsActiveEntity,code,name),relationToEntity1($select=IsActiveEntity,code,name)&$skip=0&$top=10 HTTP/1.1

{"@odata.context":"$metadata#Applications(ID=6c1133ef-27c7-40d4-8c81-a9f9620bcaab,IsActiveEntity=true)/childEntity3",
"@odata.metadataEtag":"W/\"51203-91heD+6UZAwJyYJbbpWTQQIZiHU\"",
"@odata.count":1,
"value":[{"ID":"7c09a1dc-3f69-4ab5-aeef-02449ea3322e","relationToEntity1_code":"C1","relationToEntity2_code":"A1","IsActiveEntity":true,"relationToEntity2":{"code":"A1","name":"A1","IsActiveEntity":true},"relationToEntity1":{"code":"C1","name":"C1","IsActiveEntity":true}}]}


GET Applications(ID=6c1133ef-27c7-40d4-8c81-a9f9620bcaab,IsActiveEntity=true)/childEntity2?$count=true&$select=HasActiveEntity,IsActiveEntity,code,name&$skip=0&$top=10 HTTP/1.1

{"@odata.context":"$metadata#Applications(ID=6c1133ef-27c7-40d4-8c81-a9f9620bcaab,IsActiveEntity=true)/childEntity2",
"@odata.metadataEtag":"W/\"51203-91heD+6UZAwJyYJbbpWTQQIZiHU\"",
"@odata.count":1,
"value":[{"code":"A1","name":"A1","IsActiveEntity":true,"HasActiveEntity":true}]}



GET Applications(ID=6c1133ef-27c7-40d4-8c81-a9f9620bcaab,IsActiveEntity=true)/childEntity1?$count=true&$select=HasActiveEntity,IsActiveEntity,code,name&$skip=0&$top=10 HTTP/1.1

{"@odata.context":"$metadata#Applications(ID=6c1133ef-27c7-40d4-8c81-a9f9620bcaab,IsActiveEntity=true)/childEntity1",
"@odata.metadataEtag":"W/\"51203-91heD+6UZAwJyYJbbpWTQQIZiHU\"",
"@odata.count":1,"value":[{"code":"C1","name":"C1","IsActiveEntity":true,"HasActiveEntity":true}]}



GET Applications(ID=6c1133ef-27c7-40d4-8c81-a9f9620bcaab,IsActiveEntity=true)?$select=DraftAdministrativeData,HasActiveEntity,HasDraftEntity,ID,IsActiveEntity,name&$expand=DraftAdministrativeData($select=DraftIsCreatedByMe,DraftUUID,InProcessByUser) HTTP/1.1

{"@odata.context":"$metadata#Applications(ID=6c1133ef-27c7-40d4-8c81-a9f9620bcaab,IsActiveEntity=true)",
"@odata.metadataEtag":"W/\"51203-91heD+6UZAwJyYJbbpWTQQIZiHU\"",
"ID":"6c1133ef-27c7-40d4-8c81-a9f9620bcaab","name":"app","IsActiveEntity":true,"HasActiveEntity":true,"HasDraftEntity":false,"DraftAdministrativeData":null}
```

##### Batch requests sent when clicking Save:

```
POST Applications(ID="6c1133ef-27c7-40d4-8c81-a9f9620bcaab",IsActiveEntity=false)/MyService.draftPrepare
Response:
{"@odata.context":"$metadata#Applications(ID="6c1133ef-27c7-40d4-8c81-a9f9620bcaab",IsActiveEntity=false)/MyService.draftPrepare",
"@odata.metadataEtag":"W/\"51203-91heD+6UZAwJyYJbbpWTQQIZiHU\"",
"ID":""6c1133ef-27c7-40d4-8c81-a9f9620bcaab"","name":"app","IsActiveEntity":false,"HasActiveEntity":true,"HasDraftEntity":false,"DraftAdministrativeData":
{"DraftUUID":"f304201e-6db2-411c-9371-c4600d055669","CreationDateTime":"2023-03-03T18:10:03.552Z","CreatedByUser":"nobody","DraftIsCreatedByMe":true,"LastChangeDateTime":"2023-03-03T18:10:03.552Z","LastChangedByUser":"nobody","InProcessByUser":"nobody","DraftIsProcessedByMe":true}}



POST Applications(ID="6c1133ef-27c7-40d4-8c81-a9f9620bcaab",IsActiveEntity=false)/MyService.draftActivate?$select=HasActiveEntity,HasDraftEntity,ID,IsActiveEntity,name&$expand=DraftAdministrativeData($select=DraftIsCreatedByMe,DraftUUID,InProcessByUser)
Response:
{"@odata.context":"$metadata#Applications(ID="6c1133ef-27c7-40d4-8c81-a9f9620bcaab",IsActiveEntity=false)/MyService.draftActivate",
"@odata.metadataEtag":"W/\"51203-91heD+6UZAwJyYJbbpWTQQIZiHU\"",
"ID":""6c1133ef-27c7-40d4-8c81-a9f9620bcaab"","name":"app","IsActiveEntity":true,"HasActiveEntity":true,"HasDraftEntity":false,"DraftAdministrativeData":null}


GET Applications(ID="6c1133ef-27c7-40d4-8c81-a9f9620bcaab",IsActiveEntity=true)/childEntity1?$count=true&$select=HasActiveEntity,IsActiveEntity,code,name&$skip=0&$top=10 HTTP/1.1
Response:
{"@odata.context":"$metadata#Applications(ID="6c1133ef-27c7-40d4-8c81-a9f9620bcaab",IsActiveEntity=true)/childEntity1",
"@odata.metadataEtag":"W/\"51203-91heD+6UZAwJyYJbbpWTQQIZiHU\"","@odata.count":0,"value":[]}



GET Applications(ID="6c1133ef-27c7-40d4-8c81-a9f9620bcaab",IsActiveEntity=true)/childEntity2?$count=true&$select=HasActiveEntity,IsActiveEntity,code,name&$skip=0&$top=10 HTTP/1.1
Response:
{"@odata.context":"$metadata#Applications(ID="6c1133ef-27c7-40d4-8c81-a9f9620bcaab",IsActiveEntity=true)/childEntity2",
"@odata.metadataEtag":"W/\"51203-91heD+6UZAwJyYJbbpWTQQIZiHU\"","@odata.count":0,"value":[]}


GET Applications(ID="6c1133ef-27c7-40d4-8c81-a9f9620bcaab",IsActiveEntity=true)/childEntity3?$count=true&$select=ID,IsActiveEntity,relationToEntity2_code,relationToEntity1_code&$expand=relationToEntity2($select=IsActiveEntity,code,name),relationToEntity1($select=IsActiveEntity,code,name)&$skip=0&$top=10 HTTP/1.1
Response:
{"@odata.context":"$metadata#Applications(ID="6c1133ef-27c7-40d4-8c81-a9f9620bcaab",IsActiveEntity=true)/childEntity3",
"@odata.metadataEtag":"W/\"51203-91heD+6UZAwJyYJbbpWTQQIZiHU\"",
"@odata.count":1,
"value":[{"ID":"7c09a1dc-3f69-4ab5-aeef-02449ea3322e","relationToEntity1_code":"C1","relationToEntity2_code":"A1","IsActiveEntity":true,"relationToEntity2":{"name":"","code":"","IsActiveEntity":true},"relationToEntity1":{"name":"","code":"","IsActiveEntity":true}}]}
```

### Problem
{describe the problem} 

### Fix
{describe the fix} 

### Why was it missed
{Some explanation why this issue might have been missed during normal development/testing cycle}  

### How can we avoid this
{if we don’t want to see this type of issues anymore what we should do to prevent}

**Comment by nlunets** [comment]
Hi @mengmengh2022 thanks for this, would it be possible for you to share a sample project so that i can test exactly your scenario ? a zip or a repo would be helpful :)

**Comment by mengmengh2022** [comment]
Hello @nlunets, 
thanks for having a look, I've sent you the repo via mail. 
Best Regards, Mengmeng

**Comment by nlunets** [comment]
Ok issue is identified, will provide a fix soon.

**Comment by nlunets** [comment]
Should be fixed with version [2.1.71+]

**Comment by mengmengh2022** [comment]
Thank you Nicoluas, I confirm that it is now working with 2.1.71.
